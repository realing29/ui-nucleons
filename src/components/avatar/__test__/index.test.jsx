import React from 'react';
import { act, Simulate } from 'react-dom/test-utils';
import { mount } from 'enzyme';
import { Avatar } from '../index';
import UpSVG from '@dev-dep/ui-quarks/icons/16x16/Stroked/Arrows/up';

describe('<Avatar />', () => {
  it('should renders without props', () => {
    const wrapper = mount(
      <Avatar />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle props', () => {
    const wrapper = mount(
      <Avatar
        size={40}
        imageUrl='www.images.com'
        color='brand-blue'
        textColor='white'
        title='Hello World'
        icon={UpSVG}
        bgColor='additional-teal'
        bgOpacity={0.48}
        style={{ clipPath: 'url(#fake-id)' }}
      />
    );

    expect(wrapper).toMatchSnapshot();

    wrapper.setProps({ monogram: 'WH' });

    expect(wrapper).toMatchSnapshot();

    wrapper.setProps({ color: '#f0f' });

    expect(wrapper).toMatchSnapshot();
  });

  it('should render without image', () => {
    const wrapper = mount(
      <Avatar
        size={40}
        title='Hello World'
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render without image and title', () => {
    const wrapper = mount(
      <Avatar size={64} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle wrong size', () => {
    const wrapper = mount(
      <Avatar
        size={49.92}
        title='Jason'
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle image error', () => {
    const wrapper = mount(
      <Avatar
        size={49.92}
        title='John Doe'
        imageUrl='https://www.images.com/random/'
      />
    );

    expect(wrapper.find('img')).toHaveLength(1);

    act(() => {
      Simulate.error(wrapper.find('img').getDOMNode());
    });
    wrapper.update();

    expect(wrapper.find('img')).toHaveLength(0);
  });
});
