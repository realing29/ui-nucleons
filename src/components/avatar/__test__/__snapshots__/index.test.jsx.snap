// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> should handle props 1`] = `
<Avatar
  bgColor="additional-teal"
  color="brand-blue"
  imageUrl="www.images.com"
  size={40}
  style={
    Object {
      "clipPath": "url(#fake-id)",
    }
  }
  textColor="white"
  title="Hello World"
>
  <div
    className="root size-40 color__white"
    style={
      Object {
        "--avatar-color": undefined,
        "--avatar-color-opacity": 1,
        "clipPath": "url(#fake-id)",
      }
    }
  >
    <img
      alt=""
      className="layer image"
      onError={[Function]}
      src="www.images.com"
    />
    <div
      className="layer image-overlay"
    />
  </div>
</Avatar>
`;

exports[`<Avatar /> should handle props 2`] = `
<Avatar
  bgColor="additional-teal"
  color="brand-blue"
  imageUrl="www.images.com"
  monogram="WH"
  size={40}
  style={
    Object {
      "clipPath": "url(#fake-id)",
    }
  }
  textColor="white"
  title="Hello World"
>
  <div
    className="root size-40 color__white"
    style={
      Object {
        "--avatar-color": undefined,
        "--avatar-color-opacity": 1,
        "clipPath": "url(#fake-id)",
      }
    }
  >
    <img
      alt=""
      className="layer image"
      onError={[Function]}
      src="www.images.com"
    />
    <div
      className="layer image-overlay"
    />
  </div>
</Avatar>
`;

exports[`<Avatar /> should handle props 3`] = `
<Avatar
  bgColor="additional-teal"
  color="#f0f"
  imageUrl="www.images.com"
  monogram="WH"
  size={40}
  style={
    Object {
      "clipPath": "url(#fake-id)",
    }
  }
  textColor="white"
  title="Hello World"
>
  <div
    className="root size-40 color__white"
    style={
      Object {
        "--avatar-color": undefined,
        "--avatar-color-opacity": 1,
        "clipPath": "url(#fake-id)",
      }
    }
  >
    <img
      alt=""
      className="layer image"
      onError={[Function]}
      src="www.images.com"
    />
    <div
      className="layer image-overlay"
    />
  </div>
</Avatar>
`;

exports[`<Avatar /> should handle wrong size 1`] = `
<Avatar
  size={49.92}
  title="Jason"
>
  <div
    className="root size-72 color__gray87"
    style={
      Object {
        "--avatar-color": "#f5f5f5",
        "--avatar-color-opacity": 1,
      }
    }
  >
    <span
      className="monogram"
    >
      J
    </span>
  </div>
</Avatar>
`;

exports[`<Avatar /> should render without image 1`] = `
<Avatar
  size={40}
  title="Hello World"
>
  <div
    className="root size-40 color__gray87"
    style={
      Object {
        "--avatar-color": "#f5f5f5",
        "--avatar-color-opacity": 1,
      }
    }
  >
    <span
      className="monogram"
    >
      WH
    </span>
  </div>
</Avatar>
`;

exports[`<Avatar /> should render without image and title 1`] = `
<Avatar
  size={64}
>
  <div
    className="root size-64 color__gray87"
    style={
      Object {
        "--avatar-color": "#f5f5f5",
        "--avatar-color-opacity": 1,
      }
    }
  >
    <PersonSVG
      className="icon"
      fill="#212121"
      height={24}
      width={24}
    >
      <svg
        className="icon"
        fill="#212121"
        height={24}
        viewBox="0 0 24 24"
        width={24}
      >
        <path
          clipRule="evenodd"
          d="M11.88 1C8.723 1 6.356 3.887 6.975 6.98l.455 2.274a4.66 4.66 0 009.138 0l.455-2.273C17.642 3.887 15.276 1 12.12 1h-.242zM8.936 6.588A3 3 0 0111.88 3h.242a3 3 0 012.941 3.588l-.454 2.274a2.66 2.66 0 01-5.216 0l-.455-2.274z"
          fillRule="evenodd"
        />
        <path
          d="M7.214 14.977a1 1 0 00-.428-1.954c-2.223.486-3.462 1.496-4.123 2.508a4.205 4.205 0 00-.573 1.311 3.43 3.43 0 00-.09.626c-.009.51 0 1.021 0 1.532a3 3 0 003 3h14a3 3 0 003-3c0-.51.007-1.021 0-1.532l-.002-.048a3.449 3.449 0 00-.088-.578 4.203 4.203 0 00-.573-1.31c-.66-1.013-1.9-2.023-4.123-2.509a1 1 0 10-.428 1.954c1.777.389 2.538 1.129 2.877 1.648.178.272.262.518.302.685a1.44 1.44 0 01.035.203V19a1 1 0 01-1 1H5a1 1 0 01-1-1v-1.484l.002-.025c.004-.035.013-.097.033-.18.04-.168.124-.414.302-.686.34-.52 1.1-1.26 2.877-1.648z"
        />
      </svg>
    </PersonSVG>
  </div>
</Avatar>
`;

exports[`<Avatar /> should renders without props 1`] = `
<Avatar>
  <div
    className="root size-72 color__gray87"
    style={
      Object {
        "--avatar-color": "#f5f5f5",
        "--avatar-color-opacity": 1,
      }
    }
  >
    <PersonSVG
      className="icon"
      fill="#212121"
      height={24}
      width={24}
    >
      <svg
        className="icon"
        fill="#212121"
        height={24}
        viewBox="0 0 24 24"
        width={24}
      >
        <path
          clipRule="evenodd"
          d="M11.88 1C8.723 1 6.356 3.887 6.975 6.98l.455 2.274a4.66 4.66 0 009.138 0l.455-2.273C17.642 3.887 15.276 1 12.12 1h-.242zM8.936 6.588A3 3 0 0111.88 3h.242a3 3 0 012.941 3.588l-.454 2.274a2.66 2.66 0 01-5.216 0l-.455-2.274z"
          fillRule="evenodd"
        />
        <path
          d="M7.214 14.977a1 1 0 00-.428-1.954c-2.223.486-3.462 1.496-4.123 2.508a4.205 4.205 0 00-.573 1.311 3.43 3.43 0 00-.09.626c-.009.51 0 1.021 0 1.532a3 3 0 003 3h14a3 3 0 003-3c0-.51.007-1.021 0-1.532l-.002-.048a3.449 3.449 0 00-.088-.578 4.203 4.203 0 00-.573-1.31c-.66-1.013-1.9-2.023-4.123-2.509a1 1 0 10-.428 1.954c1.777.389 2.538 1.129 2.877 1.648.178.272.262.518.302.685a1.44 1.44 0 01.035.203V19a1 1 0 01-1 1H5a1 1 0 01-1-1v-1.484l.002-.025c.004-.035.013-.097.033-.18.04-.168.124-.414.302-.686.34-.52 1.1-1.26 2.877-1.648z"
        />
      </svg>
    </PersonSVG>
  </div>
</Avatar>
`;
